_posts/2021-10-21-segmentation-torch-android/segmentation-torch-android.Rmd

```diff
@@ -0,0 +1,279 @@
+---
+title: "Train in R, run on Android: Image segmentation with torch"
+description: > 
+ We train a model for image segmentation in R, using torch together with luz, its high-level interface. We then JIT-trace the model on example input, so as to obtain an optimized representation that can run with no R installed. Finally, we show the model being run on Android.
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanasegmentationtorchandroid
+date: 10-21-2021
+bibliography: bibliography.bib
+categories:
+  - Torch
+  - Image Recognition & Image Processing
+  - R
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/segmentation_android.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+In a sense, image segmentation is not that different from image classification. It's just that instead of categorizing an image as a whole, segmentation results in a label for *every single pixel.* And as in image classification, the categories of interest depend on the task: Foreground versus background, say; different types of tissue; different types of vegetation; et cetera.
+
+The present post is not the first, on this blog, to treat that topic; and like all prior[^1] ones, it makes use of a U-Net architecture[^2] to achieve its goal. Central characteristics (of this post, not U-Net) are:
+
+[^1]: Cf. [Brain image segmentation with torch](https://blogs.rstudio.com/ai/posts/2020-11-30-torch-brain-segmentation/) (using, non-surprisingly, `torch`), and [Image segmentation with U-Net](https://blogs.rstudio.com/ai/posts/2019-08-23-unet/) (using TensorFlow/Keras).
+
+[^2]:  @RonnebergerFB15
+
+1.  It demonstrates how to perform data augmentation for an image segmentation task.
+
+2.  It uses [luz](https://mlverse.github.io/luz/), `torch`'s high-level interface, to train the model.
+
+3.  It [JIT-traces](https://blogs.rstudio.com/ai/posts/2021-08-10-jit-trace-module/) the trained model and saves it for deployment on mobile devices.
```

Comment:
I think you need to spell out the words for JIT at least once in this post. Since this is the first mention, it's the logical choice (though I know that's unappealing, given it's in a bullet _and_ a hyphenated term).

Reply:
absolutely, will do!